sw/device/lib/dif/dif_sram_ctrl.h

```diff
@@ -0,0 +1,215 @@
+  /**
+   * Indicates that the operation succeeded.
+   */
+  kDifSramCtrlOk = 0,
+  /**
+   * Indicates some unspecified failure.
+   */
+  kDifSramCtrlError = 1,
+  /**
+   * Indicates that some parameter passed into a function failed a
+   * precondition.
+   *
+   * When this value is returned, no hardware operations occured.
+   */
+  kDifSramCtrlBadArg = 2,
+  /**
+   * Indicates that this operation has been locked out, and can never
+   * succeed until hardware reset.
+   */
+  kDifSramCtrlLocked = 3,
+} dif_sram_ctrl_result_t;
+
+/**
+ * A toggle state: enabled, or disabled.
+ *
+ * This enum may be used instead of a `bool` when describing an enabled/disabled
+ * state.
+ */
+typedef enum dif_sram_ctrl_toggle {
+  /**
+   * The "enabled" state.
+   */
+  kDifSramCtrlToggleEnabled,
+  /**
+   * The "disabled" state.
+   */
+  kDifSramCtrlToggleDisabled,
+} dif_sram_ctrl_toggle_t;
+
+/**
+ * Creates a new handle for SRAM Controller.
+ *
+ * This function does not actuate the hardware.
+ *
+ * @param params Hardware instantiation parameters.
+ * @param[out] handle Out param for the initialized handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_sram_ctrl_result_t dif_sram_ctrl_init(dif_sram_ctrl_params_t params,
+                                          dif_sram_ctrl_t *handle);
+
+/**
+ * Trigger one alert event.
+ *
+ * @param handle A SRAM Controller handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_sram_ctrl_result_t dif_sram_ctrl_trigger_alert(
+    const dif_sram_ctrl_t *handle);
+
+/**
+ * Lock execution enable register.
+ *
+ * This function prevents any further writes to the
+ * execution enable register. Future attempts to enable or disable
+ * execution from SRAM will fail.
+ *
+ * @param handle A SRAM Controller handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_sram_ctrl_result_t dif_sram_ctrl_execution_set_locked(
+    const dif_sram_ctrl_t *handle);
+
+/**
+ * Reports whether the execution enable register is locked or not.
+ *
+ * If the execution enable register is locked then it is no longer possible to
+ * enable or disable execution from SRAM.
+ *
+ * @param handle A SRAM Controller handle.
+ * @param state[out] Enabled if locked, disabled if unlocked.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_sram_ctrl_result_t dif_sram_ctrl_execution_get_locked(
+    const dif_sram_ctrl_t *handle, dif_sram_ctrl_toggle_t *state);
+
+/**
+ * Reports whether the control register is locked or not.
+ *
+ * If the control register is locked then it is no longer possible to request a
+ * new scrambling key.
+ *
+ * @param handle A SRAM Controller handle.
+ * @param state[out] Enabled if locked, disabled if unlocked.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_sram_ctrl_result_t dif_sram_ctrl_control_get_locked(
+    const dif_sram_ctrl_t *handle, dif_sram_ctrl_toggle_t *state);
```

Comment:
Nit: I don't mind, but @mcy likes when non `init` routine use the name of the module instead of `handle` => s/handle/sram_ctrl/

Reply:
Done